packages/multichain-network-controller/src/utils.ts

```diff
@@ -130,3 +151,121 @@ export const toMultichainNetworkConfigurationsByChainId = (
     }),
     {},
   );
+
+// TODO: Move this function to @metamask/utils
+/**
+ * Type guard to check if a namespace is a known CAIP namespace.
+ *
+ * @param namespace - The namespace to check
+ * @returns Whether the namespace is a known CAIP namespace
+ */
+function isKnownCaipNamespace(
+  namespace: string,
+): namespace is KnownCaipNamespace {
+  return Object.values<string>(KnownCaipNamespace).includes(namespace);
+}
+
+/**
+ * Validates CAIP-10 account IDs format.
+ *
+ * @param accountIds - Array of account IDs to validate
+ * @throws Error if any account ID is invalid
+ */
+export function assertCaipAccountIds(
+  accountIds: string[],
+): asserts accountIds is CaipAccountId[] {
+  if (!accountIds.length) {
+    throw new Error('At least one account ID is required');
+  }
+
+  const invalidIds = accountIds.filter((id) => {
+    const {
+      address,
+      chain: { namespace },
+    } = parseCaipAccountId(id as CaipAccountId);
+
+    if (!isKnownCaipNamespace(namespace)) {
+      return true;
+    }
+
+    switch (namespace) {
+      case KnownCaipNamespace.Eip155:
+        return !isValidHexAddress(address as Hex);
+      case KnownCaipNamespace.Solana:
+        return !isSolanaAddress(address);
+      case KnownCaipNamespace.Bip122:
+        return !isBtcMainnetAddress(address);
```

Comment:
Do we want to only support mainnet? We are slowly re-introducing testnet on the `MultichainNetworkController`:
- https://github.com/MetaMask/core/pull/5589

So maybe we wanna add it there too? If you prefer not, I'd go with a `TODO: Add testnet support once ready` or something like this.